crates/languages/src/python.rs

```diff
@@ -427,6 +434,131 @@ impl ToolchainLister for PythonToolchainProvider {
     }
 }
 
+pub struct EnvironmentApi<'a> {
+    global_search_locations: Arc<Mutex<Vec<PathBuf>>>,
+    project_env: &'a HashMap<String, String>,
+}
+
+impl<'a> EnvironmentApi<'a> {
```

Comment:
This looks a bit like a copypaste of impl from pet itself; could we instead make `EnvironmentApi` a bit like a facade where it looks paths up in `project_env`, and if that fails, it uses `pet_core::os_environment::EnvironmentApi` (that can be it's another member)?

Reply:
Done